lib/printer.ts

```diff
@@ -717,7 +718,11 @@ function genericPrintNoParens(path: any, options: any, print: any) {
                     allowBreak = !multiLine;
                 } else if (len !== 1 && isTypeAnnotation) {
                     parts.push(separator);
-                } else if (!oneLine && util.isTrailingCommaEnabled(options, "objects")) {
+                } else if (
+                    !oneLine && 
+                    util.isTrailingCommaEnabled(options, "objects") &&
+                    childType !== 'RestElement'
```

Comment:
Inline `childType`? Unless I’m missing why it should be hoisted.

Reply:
Will do!